### PR TITLE
airthings_wave: refactor to eliminate code duplication

### PR DESCRIPTION
**Pull request by kpfleming**
# What does this implement/fix?

Refactoring of `airthings_wave_mini` and `airthings_wave_plus` components into a new `airthings_wave_base` component, eliminating code duplication.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Other (refactoring only)

## Test Environment

- [X] ESP32
- [x] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040

## Checklist:
  - [X] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

### CODE REVIEW

File: esphome/components/airthings_wave_mini/sensor.py [hunk 1]
```diff
@@ -1,82 +1,26 @@
...
     CONF_ID,
-    CONF_HUMIDITY,
-    CONF_TVOC,
-    CONF_PRESSURE,
-    CONF_TEMPERATURE,
-    UNIT_PARTS_PER_BILLION,
-    ICON_RADIATOR,
 )
 
-DEPENDENCIES = ["ble_client"]
```
Comment (jesserockz) [inline esphome/components/airthings_wave_mini/sensor.py:22]: Please leave the DEPENDENCIES here as well as they show up nicer as errors in the editor on the real components

File: esphome/components/airthings_wave_base/airthings_wave_base.cpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+void AirthingsWaveBase::request_read_values_() {
+  auto status = esp_ble_gattc_read_char(this->parent()->get_gattc_if(), this->parent()->get_conn_id(), this->handle_,
+                                        ESP_GATT_AUTH_REQ_NONE);
+  if (status) {
+    ESP_LOGW(TAG, "Error sending read request for sensor, status=%d", status);
+  }
+}
+
+AirthingsWaveBase::AirthingsWaveBase() : PollingComponent(10000) {}
```
Comment (jesserockz) [inline esphome/components/airthings_wave_base/airthings_wave_base.cpp]: This should not have actually been in the main component, safe to remove.
```suggestion
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: esphome/components/airthings_wave_mini/sensor.py (hunk 1 above)
Comment (kpfleming) [reply esphome/components/airthings_wave_mini/sensor.py:22]: Sure, I can just `import` that from the base into the real components.

File: esphome/components/airthings_wave_base/airthings_wave_base.cpp (hunk 2 above)
Comment (kpfleming) [reply esphome/components/airthings_wave_base/airthings_wave_base.cpp]: Does this mean the `PollingComponent` constructor does not need to be called, or that it should be called from the constructors of the real components?

Comment (jesserockz) [reply esphome/components/airthings_wave_base/airthings_wave_base.cpp]: It doesn't need to be called, the update interval is set inside `cg.register_component`

File: esphome/components/airthings_wave_mini/sensor.py (hunk 1 above)
Comment (kpfleming) [reply esphome/components/airthings_wave_mini/sensor.py:22]: Will push an update now.

File: esphome/components/airthings_wave_base/airthings_wave_base.cpp (hunk 2 above)
Comment (kpfleming) [reply esphome/components/airthings_wave_base/airthings_wave_base.cpp]: Will push an update now.

**Comment by kpfleming** [comment]
@jesserockz Requested changes have been made and CI is passing again. Thanks for the review!

**Review by jesserockz — APPROVED** [review]

**Pull request merged**